src/components/a-form-input/index.scss

```diff
@@ -0,0 +1,104 @@
+@import "../../styles/settings/colors";
+@import "../../styles/settings/base";
+@import "../../styles/mixins/typo";
+@import "../../styles/mixins/control-transition";
+
+.a-form-input-wrapper {
+  position: relative;
+
+  display: block;
+
+  width: 100%;
+  max-width: 100%;
+
+  &--inline {
+    display: inline-block;
+
+    width: auto;
+
+    vertical-align: middle;
+  }
+}
+
+.a-form-input {
+  @include control-transition();
+  @include typo-tag();
+
+  position: relative;
+
+  display: inline-flex;
+
+  width: 100%;
+  height: 40px;
+
+  max-width: 100%;
+
+  padding: 0 20px;
+
+  align-items: center;
+
+  color: $color-prim-gray-mine-shaft;
+  background: $color-bg--white;
+  outline: none;
+  border: $control-border-width solid $control-border-color;
+
+  appearance: none;
+
+  &::placeholder {
+    color: $color-prim-gray-dusty;
+  }
+
+  &:hover,
+  &:active,
+  &:focus {
+    &:not([disabled]):not(.a-form-input--error) {
+      border-color: $control-border-color--active;
+    }
+  }
+
+  &[disabled] {
+    background-color: $color-bg--gray;
+  }
+
+  &--error {
```

Comment:
Please write class names in full length, only use & for adjoining classes and stuff like that